pkg/agent/plugin/svidstore/disk/disk.go

```diff
@@ -0,0 +1,311 @@
+package disk
+
+import (
+	"context"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl"
+	configv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/service/common/config/v1"
+	"github.com/spiffe/spire/pkg/agent/plugin/svidstore"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/common/diskutil"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	svidstorev1 "github.com/spiffe/spire/proto/spire/plugin/agent/svidstore/v1"
+	"go.uber.org/multierr"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	pluginName = "disk"
+	attrName   = "user.spire-svid"
+)
+
+type certFile struct {
```

Comment:
small naming nit: `certFile` is being used for cert chains, private keys, and bundles. Maybe `svidFile`?